app/subscriber/src/components/front-page-gallery/FrontPageGallery.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useContent, useNavigateAndScroll } from 'store/hooks';
+import { IContentModel, Row } from 'tno-core';
+
+import * as styled from './styled';
+
+interface IFrontPageGallery extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Content array.
+   */
+  frontpages: IContentModel[];
+}
+
+export const FrontPageGallery: React.FC<IFrontPageGallery> = ({ frontpages }) => {
+  const [srcUrls, setSrcUrls] = React.useState<any[]>([]);
+  const navigateAndScroll = useNavigateAndScroll();
+  const [, { stream }] = useContent();
+
+  React.useEffect(() => {
+    setSrcUrls([]);
+    if (!!frontpages) {
+      frontpages.forEach((x) => {
+        if (x.fileReferences?.length) {
+          stream(x.fileReferences[0].path).then((result) => {
+            setSrcUrls((srcUrls) => [...srcUrls, { url: result, id: x.id }]);
```

Comment:
This line is why it's adding.  Rewrite this part instead of causing a double render.

Reply:
Try one of the following.

```
  React.useEffect(() => {
    const srcUrls = frontPages
      .filter((fp) => fp.fileReferences?.length)
      .map(async (fp) => {
        const result = await stream(fp.fileReferences![0].path);
        return { url: result, id: fp.id };
      });
    setSrcUrls(srcUrls);
  }, []);

  React.useEffect(() => {
    const srcUrls = Promise.all(
      frontPages
        .filter((fp) => fp.fileReferences?.length)
        .map(async (fp) => {
          const result = await stream(fp.fileReferences![0].path);
          return { url: result, id: fp.id };
        }),
    );
    srcUrls.then((results) => setSrcUrls(results));
  }, []);
  ```

Reply:
A link with more information about awaiting a map
https://advancedweb.hu/how-to-use-async-functions-with-array-map-in-javascript/